src/VisualStudio/Core/SolutionExplorerShim/AnalyzerReferenceManager.cs

```diff
@@ -18,15 +18,17 @@ namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplore
     [Export]
     internal class AnalyzerReferenceManager : IVsReferenceManagerUser
     {
+        private readonly IServiceProvider _serviceProvider;
         private IVsReferenceManager _referenceManager;
 
         [Import]
         private AnalyzerItemsTracker _tracker = null;
 
         [ImportingConstructor]
-        internal AnalyzerReferenceManager(SVsServiceProvider serviceProvider)
+        internal AnalyzerReferenceManager(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
         {
-            _referenceManager = serviceProvider.GetService(typeof(SVsReferenceManager)) as IVsReferenceManager;
+            _serviceProvider = serviceProvider;
```

Comment:
It's actually unnecessary to import the IServiceProvider. The global service provider is always accessible via Microsoft.VisualStudio.Shell.ServiceProvider.GlobalProvider (static)